mail_gateway/models/mail_gateway.py

```diff
@@ -113,7 +113,7 @@ def create(self, mvals):
     @tools.ormcache()
     def _get_gateway_map(self, state="integrated", gateway_type=False):
         result = {}
-        for record in self.search(
+        for record in self.sudo().search(
```

Comment:
Shouldn't you limit here by company in this search?

Reply:
No, because this map is intended to be used on the controller entry for the gateway, and the controller does not know the company until we check the gateway